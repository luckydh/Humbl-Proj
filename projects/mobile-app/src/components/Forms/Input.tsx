import { IonInput, IonItem, IonLabel } from "@ionic/react";
import React from "react";

export interface InputProps {
  label?: string;
  placeholder?: string;
  readonly?: boolean;
  disabled?: boolean;
}

const Input: React.FC<InputProps> = ({
  label,
  placeholder,
  readonly = false,
  disabled = false,
}: InputProps) => (
    <IonItem>
      {label && <IonLabel>{label}</IonLabel>}
      <IonInput placeholder={placeholder} readonly={readonly} disabled={disabled} />
    </IonItem>
  );

export default Input;
