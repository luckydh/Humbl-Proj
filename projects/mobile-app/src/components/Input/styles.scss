.input-secure {
  -webkit-text-security: disc;
}