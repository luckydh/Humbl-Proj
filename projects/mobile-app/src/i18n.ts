import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
