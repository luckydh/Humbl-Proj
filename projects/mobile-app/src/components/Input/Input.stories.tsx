import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import Input, { InputProps } from "./Input";
import withPadding from "../../utils/withPadding";
import CardPlaceHolder from "../../assets/svgs/credit-card-placeholder.svg";

export default {
  title: "Components/Input",
  component: Input,
  argTypes: {},
  decorators: [withPadding],
} as Meta;

const Template: Story<InputProps> = (args) => <Input {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  placeholder: "Placeholder Text",
};

export const WithData = Template.bind({});
WithData.args = {
  value: "I'm fine",
};

export const WithAppend = Template.bind({});
WithAppend.args = {
  append: <>$</>,
  placeholder: "Placeholder Text",
};

export const WithAppendImage = Template.bind({});
WithAppendImage.args = {
  append: <img src={CardPlaceHolder} alt="Card Placeholder" />,
  placeholder: "Placeholder Text",
};

export const WithPrepend = Template.bind({});
WithPrepend.args = {
  prepend: <>$</>,
  value: "Prepend",
  placeholder: "Placeholder Text",
};

export const WithPrependImage = Template.bind({});
WithPrependImage.args = {
  prepend: <img src={CardPlaceHolder} alt="Card Placeholder" />,
  value: "Prepend",
  placeholder: "Placeholder Text",
};
export const WithError = Template.bind({});
WithError.args = {
  hasError: true,
  value: "I have an error",
};
