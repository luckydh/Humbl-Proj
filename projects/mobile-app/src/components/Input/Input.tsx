import React, { FC, useState } from "react";
import { Capacitor } from "@capacitor/core";
import cx from "classnames";
import Label from "components/Label/Label";
import { useTranslation } from "react-i18next";
import { Message } from "../Message/Message";
import "./styles.scss";

type RefReturn =
  | string
  | ((instance: HTMLInputElement | null) => void)
  | React.RefObject<HTMLInputElement>
  | null
  | undefined;

export type InputProps = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> & {
  /** Set a jsx element after or before the input */
  append?: JSX.Element;
  ariaLabel?: string;
  prepend?: JSX.Element;
  /** Show error styling */
  hasError?: boolean;
  /** Set the input as disabled */
  isDisabled?: boolean;
  /** Set the placeholder string */
  placeholder?: string;
  /** Sets the input to type "password" and will add a show/hide toggle */
  isSecure?: boolean;
  type?: string;
  name?: string;
  value?: string | number;
  readonly?: boolean;
  register: RefReturn;
  label?: string;
  errorMessage?: string;
  maxLength?: number;
  ariaLabelValue?: string;
  inputMode?: string;
};

const isAndroid = Capacitor.getPlatform() === "android";

export const inputBaseClasses =
  "text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent";
export const wrapperBaseClasses =
  "py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none";

export const Input: FC<InputProps> = ({
  append,
  ariaLabel,
  prepend,
  label,
  inputMode,
  hasError = false,
  isDisabled = false,
  placeholder,
  register,
  type = "text",
  value,
  name,
  isSecure,
  readOnly,
  className,
  errorMessage,
  maxLength,
  ariaLabelValue,
  ...rest
}) => {
  const { t } = useTranslation();
  const classes = cx(
    wrapperBaseClasses,
    isDisabled ? "disabled" : "",
    append ? "pr-4" : "",
    prepend ? "pl-4 " : "",
    hasError ? "border-red" : "",
    errorMessage ? "mb-2" : "",
    className
  );

  const [showText, setShowText] = useState(false);

  const textHidden = isSecure && !showText;
  const inputType = textHidden && !isAndroid ? "password" : type;

  const inputClasses = cx(inputBaseClasses, { "input-secure": textHidden && isAndroid });

  return (
    <div className="mb-2">
      {label && <LabelContainer ariaLabel={ariaLabel && `${ariaLabel}_LABEL`} label={label} />}
      {/* Note this label is purposefully not directly on the input b/c it will get swallowed by Android */}
      <div className={classes} aria-label={ariaLabelValue && `${ariaLabelValue}_FIELD`}>
        {!!prepend && <StyleInjector>{prepend}</StyleInjector>}
        <input
          className={inputClasses}
          placeholder={placeholder}
          type={inputType}
          ref={register}
          inputMode={inputMode}
          name={name}
          maxLength={maxLength}
          value={value}
          style={{ height: 48, fontSize: 20 }}
          readOnly={readOnly}
          {...rest}
        />
        {!!append && <StyleInjector>{append}</StyleInjector>}
        {isSecure && (
          <StyleInjector>
            <div
              aria-label={ariaLabel && `${ariaLabel}_SHOW_INPUT`}
              onClick={() => setShowText(!showText)}
              className="text-xs mr-2">
              {showText ? t("input.hide") : t("input.show")}
            </div>
          </StyleInjector>
        )}
      </div>
      {errorMessage && (
        <div>
          <Message ariaLabel={ariaLabel && `${ariaLabel}ALERT_LABEL`} variant="error">
            {errorMessage}
          </Message>
        </div>
      )}
    </div>
  );
};

export default Input;

Input.displayName = "Input";

export const LabelContainer: FC<{ label: string; ariaLabel?: string }> = ({ label, ariaLabel }) => (
  <div className="ml-2">
    <Label ariaLabel={ariaLabel}>{label}</Label>
  </div>
);

const StyleInjector: React.FC = ({ children }) => (
  <div className="max-h-8 flex" style={{ maxWidth: "3.5rem" }}>
    {children}
  </div>
);
