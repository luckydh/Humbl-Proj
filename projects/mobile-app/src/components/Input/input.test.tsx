import React from "react";
import { StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Input from "./Input";
//Disable suspense on menu drawer
import i18n from "../../i18n";

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}

it("renders Input field correctly", async () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Input register={() => null} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
