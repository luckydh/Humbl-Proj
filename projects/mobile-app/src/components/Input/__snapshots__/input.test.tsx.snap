// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input field correctly 1`] = `
<div
  className="mb-2"
>
  <div
    className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
  >
    <input
      className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
      style={
        Object {
          "fontSize": 20,
          "height": 48,
        }
      }
      type="text"
    />
  </div>
</div>
`;
